src/FakeItEasy/Creation/DefaultFakeCreatorFacade.cs

```diff
@@ -35,6 +35,14 @@ public T CreateFake<T>(Action<IFakeOptions<T>> optionsBuilder)
             return (T)this.fakeAndDummyManager.CreateFake(typeof(T), options => optionsBuilder((IFakeOptions<T>)options));
         }
 
+        public object CreateFake(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
+        {
+            Guard.AgainstNull(typeOfFake, nameof(typeOfFake));
+            Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));

```

Comment:
This method does not de-reference either argument. These guards should only be added if it's not possible for a method further down the call stack to give the same quality of exception message. Has that been verified?